From b592a62011d49b7dce7f08cd67912d86bbc9265e Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@chainguard.dev>
Date: Fri, 7 Nov 2025 06:13:29 +0000
Subject: [PATCH] systemd.fc: allow generators to create early and late units

Allow all generators to create files in all output directories,
including early and late.

References:
- https://www.freedesktop.org/software/systemd/man/latest/systemd.generator.html#Output%20directories
---
 policy/modules/system/systemd.fc | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/policy/modules/system/systemd.fc b/policy/modules/system/systemd.fc
index fc8e09817..a238f9fcd 100644
--- a/policy/modules/system/systemd.fc
+++ b/policy/modules/system/systemd.fc
@@ -132,7 +132,9 @@ HOME_DIR/\.config/systemd/user(/.*)?		gen_context(system_u:object_r:systemd_unit
 /run/.*nologin.*		gen_context(system_u:object_r:systemd_logind_var_run_t,s0)
 /run/systemd/default-hostname	--	gen_context(system_u:object_r:hostname_etc_t,s0)
 
-/run/systemd/generator(/.*)?	gen_context(system_u:object_r:systemd_generator_unit_file_t,s0)
+/run/systemd/generator(/.*)?		gen_context(system_u:object_r:systemd_generator_unit_file_t,s0)
+/run/systemd/generator.early(/.*)?	gen_context(system_u:object_r:systemd_generator_unit_file_t,s0)
+/run/systemd/generator.late(/.*)?	gen_context(system_u:object_r:systemd_generator_unit_file_t,s0)
 
 /run/systemd/io\.systemd\.Login	-s	gen_context(system_u:object_r:systemd_logind_var_run_t,s0)
 /run/systemd/io\.systemd\.ManagedOOM	-s	gen_context(system_u:object_r:systemd_oomd_var_run_t,s0)
-- 
2.51.0

